### PR TITLE
WT-3369 WT_CURSOR->uri should always match the URI used to open the cursor

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
retest this please

File: src/cursor/cur_table.c [hunk 1]
```diff
@@ -988,6 +988,13 @@ __wt_curtable_open(WT_SESSION_IMPL *session,
 		    table->cgroups[0]->source, NULL, cfg, cursorp);
 
 		__wt_schema_release_table(session, table);
+		if (ret == 0) {
+			/* Fix up the public URI to match what was passed in. */
+			cursor = *cursorp;
+			__wt_free(session, cursor->uri);
+			cursor->uri = NULL;
```
Comment (michaelcahill) [inline src/cursor/cur_table.c]: This is redundant -- `__wt_free` is actually a macro that sets the pointer to NULL.

Comment (ddanderson) [inline src/cursor/cur_table.c]: @michaelcahill, thanks.  I've fixed that, and also another issue with LSM ownership of the `WT_CURSOR->uri` field.

Can you take another look please?

**Comment by michaelcahill** [comment]
Thanks @ddanderson, lgtm.  I'll merge.

**Pull request merged**